lib/insights/api/common/application_controller_mixins/exception_handling.rb

```diff
@@ -35,6 +38,24 @@ def error_code_from_class(exception)
               DEFAULT_ERROR_CODE
             end
           end
+
+          def api_client_exception?(exc)
+            exc.respond_to?(:code) && exc.respond_to?(:response_body) && exc.respond_to?(:response_headers) &&
+              !exc.response_body.nil?
```

Comment:
Should `!exc.response_body.nil?` check be here? ApiException itself allows response_body to be nil. Better move it into method `api_client_errors`

Reply:
@hsong-rh I was doing that initially in the api_client_errors the reason I moved it here is because if the body is nil then we want to treat it as a regular exception and add a single error object.